go/vt/vtgate/engine/join_values_test.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2025 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package engine
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"vitess.io/vitess/go/sqltypes"
+	querypb "vitess.io/vitess/go/vt/proto/query"
+)
+
+func TestJoinValuesExecute(t *testing.T) {
+
+	/*
+		select col1, col2, col3, col4, col5, col6 from left join right on left.col1 = right.col4
+		LHS: select col1, col2, col3 from left
+		RHS: select col5, col6, id from (values row(1,2), ...) left(id,col1) join right on left.col1 = right.col4
+	*/
```

Comment:
nit: the example query does not have `id` column but RHS has in the select.
output shows `col4` but it is not present in LHS or RHS